File: structural/typescript/adapter_structured/interfaces.ts
export interface MediaPlayer {
  play(audioType: string, fileName: string): void;
}

export interface PaymentProcessor {
  processPayment(amount: number): void;
}

export interface CelsiusTemperature {
  getTemperatureInCelsius(): number;
  setTemperatureInCelsius(temp: number): void;
}

export interface FahrenheitTemperature {
  getTemperatureInFahrenheit(): number;
  setTemperatureInFahrenheit(temp: number): void;
}

File: structural/dart/adapter_structured/interfaces.dart
abstract class MediaPlayer {
  void play(String audioType, String fileName);
}

abstract class PaymentProcessor {
  void processPayment(double amount);
}

abstract class CelsiusTemperature {
  double getTemperatureInCelsius();
  void setTemperatureInCelsius(double temp);
}

abstract class FahrenheitTemperature {
  double getTemperatureInFahrenheit();
  void setTemperatureInFahrenheit(double temp);
}
